File: package/main/src/String/countOccurrences.ts
/**
 * Counts the number of non-overlapping occurrences of a substring within a
 * string. Returns 0 when the search string is empty.
 * @param {string} string_ - Input string
 * @param {string} search - Substring to count
 * @returns {number} Number of occurrences
 * @example
 * countOccurrences("ababab", "ab"); // 3
 * countOccurrences("aaaa", "aa"); // 2
 */
export const countOccurrences = (string_: string, search: string): number => {
  if (search === "") {
    return 0;
  }
  let count = 0;
  let index = string_.indexOf(search);
  while (index !== -1) {
    count++;
    index = string_.indexOf(search, index + search.length);
  }
  return count;
};

File: package/umt_python/src/string/count_occurrences.py
def count_occurrences(string_: str, search: str) -> int:
    """
    Counts the number of non-overlapping occurrences of a substring within a
    string. Returns 0 when the search string is empty.

    Args:
        string_: Input string.
        search: Substring to count.

    Returns:
        Number of occurrences.

    Example:
        >>> count_occurrences("ababab", "ab")
        3
        >>> count_occurrences("aaaa", "aa")
        2
        >>> count_occurrences("banana", "a")
        3
        >>> count_occurrences("hello", "")
        0
    """
    if search == "":
        return 0
    count = 0
    index = string_.find(search)
    while index != -1:
        count += 1
        index = string_.find(search, index + len(search))
    return count
